pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1265,6 +1265,32 @@ def compute_matrix() -> np.ndarray:  # pylint: disable=arguments-differ
         """
         return np.array([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
 
+    @staticmethod
+    @lru_cache()
+    def compute_sparse_matrix() -> sparse.csr_matrix:  # pylint: disable=arguments-differ
+        r"""Sparse Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.SWAP.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.SWAP.compute_matrix())
```

Comment:
Should this be this instead?
```suggestion
        >>> print(qml.SWAP.compute_sparse_matrix())
```